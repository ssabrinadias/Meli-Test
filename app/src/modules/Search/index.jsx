import React, { useState, useEffect } from "react";
import styled from 'styled-components'
import { theme } from 'styled-tools'
import Language from "../../components/language" 
import IconSearch from "../../assets/images/ic_Search_2x.png"
import { Redirect } from 'react-router'


const Content = styled.div`
    display: flex;
`

const Form = styled.form ` 
    display: flex;
    width: 100%;
`

const Input = styled.input`
    height: 40px;
    padding: 10px;
    width: 100%;
    background: #fff;
    border: none;
    border-radius: 4px 0px 0px 4px;
`

const Button = styled.button`
    height: 40px;
    width: 50px;
    border: none;
    background: #FFF;
    border-radius: 0px 4px 4px 0px;
    background-image: url(${IconSearch});
    background-repeat: no-repeat;
    background-size: 22px;
    background-position: center;
    background-color: ${theme('colors.light.main')};
`

const Search = () => {
    
    const [search, setSearch] = useState("")
    const [fireRedirect, setFire] = useState(null)
    
    const handleSubmit = (evt)=>{
        evt.preventDefault()
        setFire(search)
    }
    
    return (
        <Form onSubmit={handleSubmit}>
            <Input
                value={search}
                onChange={e => setSearch(e.target.value)}
                placeholder={Language.ARS.placeholder.Search}
                type="text"
                name="search"
                required
            />
            {fireRedirect && <Redirect to={`/items?search=${search}`} />}
           
            <Button type="submit"></Button>
        </Form> 
    )
};
export default Search;