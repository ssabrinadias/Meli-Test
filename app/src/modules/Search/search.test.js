import React from "react";
import { render } from "@testing-library/react";
import Search from ".";

describe("Search component", () => {
	it("[snapshot] - should be able to render ", () => {
		const { container } = render(<Search />);
		expect(container).toMatchSnapshot();
	});
});
