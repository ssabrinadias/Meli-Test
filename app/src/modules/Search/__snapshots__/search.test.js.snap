// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component [snapshot] - should be able to render  1`] = `
<div>
  <form
    class="sc-AxirZ hvoVdz"
  >
    <input
      class="sc-AxiKw iculOt"
      name="search"
      placeholder="Nunca dejes de Buscar"
      required=""
      type="text"
      value=""
    />
    <button
      class="sc-AxhCb kBUfZc"
      type="submit"
    />
  </form>
</div>
`;
